cache.go

```diff
@@ -318,8 +432,213 @@ func (c *GoBuildCache) ToRunOption(distroKey string, opts ...GoBuildCacheOption)
+
+		// Only download and set up precompiled sccache for distros that need it or Windows
+		if needsPrecompiled[distroName] || isWindows {
+			var binaryPath string
+			if isWindows {
+				binaryPath = sccacheBinaryWin
+			} else {
+				binaryPath = sccacheBinary
+			}
+
+			// Download sccache binary using SourceHTTP during build phase
+			sccacheSource := GetSccacheSource(platform)
+			sccacheState, err := sccacheSource.HTTP.AsState("sccache", llb.Platform(*platform))
+			if err != nil {
+				// If we can't get sccache, continue without it
+				return
+			}
+
+			// Create a setup command to extract and configure sccache
+			var setupCmd string
+			if isWindows {
+				// For Windows, extract the archive and copy the binary
+				setupCmd = fmt.Sprintf(`
+if (Test-Path "/tmp/dalec/sccache") {
+	# Create temporary extraction directory
+	New-Item -Path "/tmp/dalec/sccache-extract" -ItemType Directory -Force
+	# Extract sccache from tar.gz - the archive contains a versioned directory
+	tar -xzf /tmp/dalec/sccache -C /tmp/dalec/sccache-extract/
+	# Find the sccache.exe file and copy it (it's in a subdirectory like sccache-v0.10.0-x86_64-pc-windows-msvc/)
+	$sccacheExe = Get-ChildItem -Path "/tmp/dalec/sccache-extract/" -Name "sccache.exe" -Recurse | Select-Object -First 1
+	if ($sccacheExe) {
+		Copy-Item -Path $sccacheExe.FullName -Destination "%s" -Force
+		$env:RUSTC_WRAPPER = "%s"
+		Write-Host "Using downloaded sccache for cargo build caching"
+	} else {
+		Write-Host "Warning: sccache.exe not found in archive"
+	}
+} elseif (Get-Command sccache -ErrorAction SilentlyContinue) {
+	$env:RUSTC_WRAPPER = "sccache"
+	Write-Host "Using system sccache for cargo build caching"
+} else {
+	Write-Host "Warning: sccache not available, cargo build caching disabled"
+}`, binaryPath, binaryPath)
+			} else {
+				// For Linux, extract the archive and copy the binary
+				setupCmd = fmt.Sprintf(`
+if [ -f /tmp/dalec/sccache ]; then
+	# Create temporary extraction directory
+	mkdir -p /tmp/dalec/sccache-extract
+	# Extract sccache from tar.gz - the archive contains a versioned directory like sccache-v0.10.0-x86_64-unknown-linux-musl/
+	tar -xzf /tmp/dalec/sccache -C /tmp/dalec/sccache-extract/
+	# Find the sccache binary and copy it
+	sccache_bin=$(find /tmp/dalec/sccache-extract/ -name "sccache" -type f | head -1)
+	if [ -n "$sccache_bin" ]; then
+		cp "$sccache_bin" "%s" && chmod +x "%s"
+		export RUSTC_WRAPPER="%s"
+		echo "Using downloaded sccache for cargo build caching"
+	else
+		echo "Warning: sccache binary not found in archive"
+	fi
+elif command -v sccache >/dev/null 2>&1; then
+	export RUSTC_WRAPPER=sccache
+	echo "Using system sccache for cargo build caching"
+else
+	echo "Warning: sccache not available, cargo build caching disabled"
+fi`, binaryPath, binaryPath, binaryPath)
+			}
+
+			// Mount the sccache source and add the setup command
+			llb.AddMount("/tmp/dalec/sccache", sccacheState, llb.SourcePath("sccache")).SetRunOption(ei)
```

Comment:
Instead let's do something like `ei.State.Run(extractCommand)` and extract sccacache directly to an llb.State that we can mount without needin to pass through the extra stuff through the run option.

e.g.


```go
ei.State.Run(
  llb.AddMount("/tmp/dalec/sccache", sccacheState, llb.SourcePath(llb.SourcePath("sccache")
 ShArgs("<extract command here>")
).AddMount("/path/to/extractOutput", llb.Scratch())
```